### PR TITLE
NET-7178: add workload selector  to MeshGateway protobuf definition

### PR DESCRIPTION
**Pull request by NiniOak**
### Description

This PR updates the MeshGateway protobuf definition to include workload selector. Another [PR](https://github.com/hashicorp/consul-k8s/pull/3465) in Consul-K8s will build upon this to use this field but not add it to the meshgateway CRD

### Testing & Reproduction steps
UI test passes
No linting errors

MeshGateway struct updates only. Validate the fields are updated correctly and CI/Lint pass

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: proto-public/pbmesh/v2beta1/mesh_gateway.proto [hunk 1]
```diff
@@ -15,6 +16,9 @@ message MeshGateway {
 
   // +kubebuilder:validation:MinItems=1
   repeated MeshGatewayListener listeners = 2;
+
+  // Selection of workloads to be exposed through the MeshGateway
+  hashicorp.consul.catalog.v2beta1.WorkloadSelector workloads = 3;
```
Comment (sarahalsmiller) [inline proto-public/pbmesh/v2beta1/mesh_gateway.proto:21]: Hmm based on Nathan's comment [here](https://github.com/hashicorp/consul-k8s/pull/3465#pullrequestreview-1815811800), I think you might need to break this out so it doesn't show up on the CRD. I couldn't find a kubetag that ignores a field completely, but it does look like you can set it to [ignore unexported fields ](https://book.kubebuilder.io/reference/controller-gen)

Comment (missylbytes) [reply proto-public/pbmesh/v2beta1/mesh_gateway.proto]: Does that mean we would need to add the ignore option on the controller-gen command itself?

File: proto-public/pbmesh/v2beta1/mesh_gateway.proto [hunk 2]
```diff
@@ -15,6 +16,9 @@ message MeshGateway {
 
   // +kubebuilder:validation:MinItems=1
   repeated MeshGatewayListener listeners = 2;
+
+  // Selection of workloads to be exposed through the MeshGateway
```
Comment (nathancoleman) [inline proto-public/pbmesh/v2beta1/mesh_gateway.proto:20]: Stole the second line from other usages of this type across consul

```suggestion
  // Selection of workloads to be configured as mesh gateways.
  // These can be prefixes or specific workload names.
```

File: proto-public/pbmesh/v2beta1/mesh_gateway.proto (hunk 1 above)
Comment (nathancoleman) [reply proto-public/pbmesh/v2beta1/mesh_gateway.proto:21]: I also couldn't find an option to ignore a single field; however, we paired as a group earlier today and worked through omitting the field in the consul-k8s CRD. We'll just use a different parent struct for the CRD in consul-k8s and then map that to a `MeshGateway` defined here by adding a workload selector before syncing into consul

**Review by nathancoleman — APPROVED** [review]
Nice work! Some optional commentary below but fine with this merging

**Pull request merged**